app/controllers/concerns/traceable.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+require 'tracers/datadog_adapter'
+
+# Provides functionality to controllers for tagging them with specific services.
+# This tagging allows monitoring and tracing systems (like Datadog) to identify and group
+# the activities of specific controllers based on the service tags they are associated with.
+#
+# @example Setting a service tag for a controller
+#   class MyController < ApplicationController
+#     include Traceable
+#     service_tag :my_service
+#   end
+#
+# @note The actual interaction with the monitoring service is abstracted away
+#   and is handled by {Tracers::DatadogAdapter}. This module mainly focuses on
+#   providing the mechanism to set and validate the service tags for controllers.
+#
+# @see Tracers::DatadogAdapter for details on how tags are set for external services.
+module Traceable
+  extend ActiveSupport::Concern
+
+  # @!attribute [r] TRACE_ADAPTER
+  #   @return [Tracers::DatadogAdapter] Adapter for interacting with external tracing services.
+  TRACE_ADAPTER = Tracers::DatadogAdapter
+
+  # @!attribute [r] SERVICES
+  #   List of valid services. These define possible service tags that a controller can have.
+  #   TODO: load this from a config file.
+  #   @return [Array<Symbol>]
+  SERVICES = %i[evss_letters evss_gi_bill_status prescriptions secure_messaging].freeze
```

Comment:
I'm not sure what to do here - I want to be mindful of avoiding developers need to define a service in **too** many places (config file, plus add annotation to controller, plus add to datadog service catalog). Unless config file == datadog service catalog? Or we leave this freeform and have an out of band process for auditing controller service tags that don't have an entry in the catalog.

@acrollet any thoughts?

Reply:
I tend to agree, from a DX perspective I prefer to either pull this list from the service catalog (perhaps even a deploy task that calls the API and populates an env var?) or allow freeform tagging. Or a middle way could be to have a CI job that runs on PRs and checks that tags found have a service catalog entry.

Reply:
I like enforcement by CI. I don't want to couple deployment with an external API call.

Reply:
I created https://github.com/department-of-veterans-affairs/va.gov-team/issues/67031 to track this need, please review/update as you see fit 🙂